### DISCUSSION TITLE
Padding after numbered heading

### DISCUSSION THREAD

**Discussion by aaronschiff** [typst/typst#4503] [Q&A] [answered]
I'm new to Typst so please forgive me for asking a basic question. I'm trying to format numbered headings so they have a gap (padding) below them. Below is what I've come up with and it works, but it seems cumbersome. Is this the best way to do it? 

```{typst}
#set heading(numbering: "1.")
#show heading: it => [
  #set text(24pt, weight: "light")
  #pad(
    bottom: 6pt,
    counter(heading).display(
       it.numbering
     ) + " " +
     it.body
  )
]
```

**Comment by laurmaedje** [comment, marked as answer]
I believe what you're looking for is `show heading: set block(below: ..)`. All block-level elements can have their `above` and `below` spacing adjusted in this way.

**Reply by aaronschiff** [reply]
Thanks! That works, although not when there is a list or bullets directly below the heading. What should I do in that case? 

<img width="341" alt="Screenshot 2024-07-05 at 21 45 26" src="https://github.com/typst/typst/assets/10150126/9cbcfc3a-de7e-42fc-9d45-46484162459d">

**Reply by laurmaedje** [reply]
This is a bug that was fixed recently: https://github.com/typst/typst/pull/4396

It should only occur if there is no blank line between the heading and list.

**Reply by aaronschiff** [reply]
Thanks again. Yes adding a blank line after the heading fixes it. Really appreciate your quick responses!
